### PR TITLE
Added test cases for enums

### PR DESCRIPTION
**Pull request by derrabus**
By coincidence, we already support enums in annotations. Let's add some tests to make sure it stays that way.

### CODE REVIEW

**Review by malarzm — DISMISSED** [review]
Do we want to advettise the fact in docs?

File: tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php [hunk 1]
```diff
@@ -295,4 +298,29 @@ public function testFunctionAnnotation(): void
...
+        self::assertSame($expectedValue, $annotation->suit);
+    }
+
+    /**
+     * @return list<array{string, Suit}>
+     */
+    public function provideEnumProperties(): array
+    {
+        return [
+            ['annotationWithDefaults', Suit::Hearts],
```
Comment (malarzm) [inline tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php]: Sadly you can't use enums here as PHP will choke on parsing file when running phpunit on older versions

Comment (derrabus) [reply tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php]: I'm surprised the data provider is executed. I'll fix that.

**Comment by derrabus** [comment]
I don't know if it's worth mentioning enums in the docs. 🤷🏻‍♂️

**Review by stof — APPROVED** [review]

File: tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php (hunk 1 above)
Comment (malarzm) [reply tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php]: PHP breaks on parsing the file, not during execution :)

Comment (derrabus) [reply tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php]: No, the file is parsed just fine as you can tell by the green test suite.

Comment (malarzm) [reply tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php]: Oh, that's actually good to know! I was basing on my last experience with doctrine/common where I hid actual enum usage as much as possible, must have drawn wrong conclusions :)

Comment (stof) [reply tests/Doctrine/Tests/Common/Annotations/AnnotationReaderTest.php]: parsing the file containing the enum is what fails. So as long as you don't trigger autoloading for it, you're fine.

**Pull request merged**